src/lib.rs

```diff
@@ -3828,6 +3838,33 @@ impl Connection {
         self.closed
     }
 
+    /// Returns true if the connection is closing.
+    #[inline]
+    pub fn is_closing(&self) -> bool {
+        self.local_error.is_some()
+    }
+
+    /// Returns the error code and reason phrase from a connection close or
+    /// application close event received from the peer, if any.
+    ///
+    /// The values contained in the tuple are symmetric with the [`close()`]
+    /// method.
+    ///
+    /// Note that a `Some` return value does not necessarily imply
+    /// [`is_closed()`] or any other connection state.
+    ///
+    /// [`close()`]: struct.Connection.html#method.close
+    /// [`is_closed()`]: struct.Connection.html#method.is_closed
+    #[inline]
+    pub fn received_close_info(&self) -> Option<(bool, u64, &[u8])> {
```

Comment:
Why not just return the `ConnectionClose` structure rather than a tuple? (that is, `ConnectionClose` would need to be a public type).